api/src/middleware/assessmentsDummyRouter.ts

```diff
@@ -0,0 +1,291 @@
+          {
+            id: 10,
+            response:
+              'Benefits of using React include its modular and reusable components, efficient updates with virtual DOM, JSX syntax, and active community.',
+            assessment_id: 1,
+            submission_id: 2,
+            question_id: 10,
+          },
+        ],
+      },
+    ];
+    let insertedAssessmentSubmissionsRow: AssessmentSubmissionRow;
+    try {
+      insertedAssessmentSubmissionsRow = await insertToAssessmentSubmissions(
+        dummyAssessmentSubmissionData[0].assessment_id,
+        dummyAssessmentSubmissionData[0].principal_id,
+        dummyAssessmentSubmissionData[0].assessment_submission_state_id,
+        dummyAssessmentSubmissionData[0].score,
+        dummyAssessmentSubmissionData[0].opened_at,
+        dummyAssessmentSubmissionData[0].submitted_at,
+        dummyAssessmentSubmissionData[0].responses
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+
+    res.status(201).json(itemEnvelope(insertedAssessmentSubmissionsRow));
+  }
+);
+
+assessmentsDummyRouter.get(
+  '/makeFacilitator/:programId/:participantId',
+  async (req, res, next) => {
+    const { programId, participantId } = req.params;
+    // a role ID of 2 corresponds to a Facilitator
+    const roleId = 2;
+
+    const programIdParsed = Number(programId);
+    const participantIdParsed = Number(participantId);
+    if (
+      !Number.isInteger(programIdParsed) ||
+      programIdParsed < 1 ||
+      !Number.isInteger(participantIdParsed) ||
+      participantIdParsed < 1
+    ) {
+      next(
+        new BadRequestError(
+          `"${programIdParsed}""${participantIdParsed}"  is not a valid program id .`
+        )
+      );
+      next(
+        new BadRequestError(
+          `"${participantIdParsed}"  is not a valid participant id .`
+        )
+      );
+      return;
+    }
+
+    let insertedProgramFacilitatorsRow: ProgramParticipantsRow;
+    try {
+      insertedProgramFacilitatorsRow = await insertToProgramParticipants(
+        participantIdParsed,
+        programIdParsed,
+        roleId
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+    res.status(201).json(itemEnvelope(insertedProgramFacilitatorsRow));
+  }
+);
+interface AssessmentStartRow {
+  id: number;
+  assessment_id?: number;
+  principal_id?: number;
+}
+assessmentsDummyRouter.get(
+  '/startAssessment/:assessmentId/:participantId',
+  async (req, res, next) => {
+    const { assessmentId, participantId } = req.params;
+
+    const participantIdParsed = Number(participantId);
+    const assessmentIdParsed = Number(assessmentId);
+
+    if (
+      !Number.isInteger(participantIdParsed) ||
+      participantIdParsed < 1 ||
+      !Number.isInteger(assessmentIdParsed) ||
+      assessmentIdParsed < 1
+    ) {
+      next(
+        new BadRequestError(
+          `"${participantIdParsed}" "${assessmentIdParsed}" is not a valid participant id or assessment id.`
```

Comment:
This is not a valid compromise. We cannot just say this, because we can achieve this branch (as in, we could be calling `BadRequestError` if either the `participantId` *or* the `assessmentId` is valid (but of course, not both), but here you're claiming that valid data is not valid. See above comments about not collapsing these tests.